Points Filter added

TheTimmaeh opened the pull request:
Finally it worked!

Used latest 'release', added points filter.

Commit:
Update sgv2-filter.user.js

Added point filter [REQ3]

Commit:
Update sgv2-filter.user.js

removed unneccessary stuff

Commit:
Reset

Commit:
Points filter added

Commit:
Fixes

TheTimmaeh commented:
I've reset the file and added the points filter + fixes ;)

Commit:
Fixed onchange function for maxPointsInput

When value is changed to higher number of points than default max, value will now be changed to default max instead of previous value.

GarionCZ commented on sgv2-filter.user.js:
```diff
@@ -470,6 +508,78 @@ function createFilterUiFilterOptionsRow() {
+    // Filter out invalid values
+    var minPointsToDisplayInputValue = parseInt(minPointsToDisplayInput.value);
+    if (minPointsToDisplayInputValue < 0 || minPointsToDisplayInputValue > 150 || minPointsToDisplayInputValue > maxPointsToDisplay) {
+      minPointsToDisplayInput.value = minPointsToDisplay;
```
I don't really like that the minPoints input and maxPoints input are handled differently (different condition structure) and therefore behave differently. I'd suggest choosing one way and do it like that. I'd also suggest it to be the same behaviour as the level range has.

GarionCZ commented on sgv2-filter.user.js:
```diff
@@ -470,6 +508,78 @@ function createFilterUiFilterOptionsRow() {
+  maxPointsToDisplayInput.onchange = function() {
+    // Filter out invalid values
+    var maxPointsToDisplayInputValue = parseInt(maxPointsToDisplayInput.value);
+    if (maxPointsToDisplayInputValue < 0 || maxPointsToDisplayInputValue < minPointsToDisplay) {
```
See comment at line 522.

GarionCZ commented on sgv2-filter.user.js:
```diff
@@ -470,6 +508,78 @@ function createFilterUiFilterOptionsRow() {
+      // If the value changed, save it and update the UI
+      GM_setValue(KEY_MIN_POINTS_TO_DISPLAY, minPointsToDisplayInputValue);
+      minPointsToDisplay = minPointsToDisplayInputValue;
+      updateFilterCaption();
```
You should also add the points to the caption (so the updateFilterCaption call actually has any effect), I'd suggest the caption to look like "Giveaway Filter (Level 0-10, Max 200 entries, X-Y points, Group GAs excluded)".

GarionCZ commented on sgv2-filter.user.js:
```diff
@@ -470,6 +508,78 @@ function createFilterUiFilterOptionsRow() {
+  minPointsToDisplayInput.onchange = function() {
+    // Filter out invalid values
+    var minPointsToDisplayInputValue = parseInt(minPointsToDisplayInput.value);
+    if (minPointsToDisplayInputValue < 0 || minPointsToDisplayInputValue > 150 || minPointsToDisplayInputValue > maxPointsToDisplay) {
```
Use the DEFAULT_MAX_POINTS_TO_DISPLAY constant instead of a hardcoded 150 value. When this will change, we'll just change the constant value and it'll work again.

GarionCZ commented on sgv2-filter.user.js:
```diff
@@ -470,6 +508,78 @@ function createFilterUiFilterOptionsRow() {
+    var maxPointsToDisplayInputValue = parseInt(maxPointsToDisplayInput.value);
+    if (maxPointsToDisplayInputValue < 0 || maxPointsToDisplayInputValue < minPointsToDisplay) {
+      maxPointsToDisplayInput.value = maxPointsToDisplay;
+    } else if (maxPointsToDisplayInputValue > 150) {
```
Use the DEFAULT_MAX_POINTS_TO_DISPLAY constant instead of a hardcoded 150 value. When this will change, we'll just change the constant value and it'll work again.

GarionCZ commented on sgv2-filter.user.js:
```diff
@@ -470,6 +508,78 @@ function createFilterUiFilterOptionsRow() {
+    
+  // Create and add the points filter
+  var showPointsSpan = document.createElement("span");
+  showPointsSpan.appendChild(document.createTextNode("Show points:"));
```
Maybe "Points to enter:" to be more descriptive.

Commit:
Updated as requested (see comments)

Changed behaviour of level & points filter, added caption update, renamed points filter in UI

Commit:
Changed comment position

GarionCZ commented on sgv2-filter.user.js:
```diff
@@ -410,9 +448,11 @@ function createFilterUiFilterOptionsRow() {
-    if (minLevelToDisplayInputValue < 0 || minLevelToDisplayInputValue > 10 || minLevelToDisplayInputValue > maxLevelToDisplay) {
-      minLevelToDisplayInput.value = minLevelToDisplay;
-    } else if (minLevelToDisplay != minLevelToDisplayInputValue) {
+    if (minLevelToDisplayInputValue < DEFAULT_MIN_LEVEL_TO_DISPLAY) {
```
I don't think it works now :) You set the desired value to the minLevelToDisplayInput.value, but you don't save i to the minLevelToDisplayInput and to the GM_setValue. I think you need to separate the last "else if" and make it just if. The two "checking" conditions should then save it not only to the minLevelToDisplayInput.value, but to the minLevelToDisplayInputValue as well.

Commit:
Fixed onchange function for level & points filter
